modules/contentbox/models/modules/ModuleService.cfc

```diff
@@ -180,9 +180,9 @@ component extends="cborm.models.VirtualEntityService" accessors="true" singleton
 	 * @name The name of the module to register
 	 * @type The type of module: core or custom
 	 */
-	Module function registerNewModule( required name, required type ){
-		var thisPath 			= variables[ arguments.type & "ModulesPath" ];
-		var thisInvocationPath 	= variables[ arguments.type & "ModulesInvocationPath" ];
+	Module function registerNewModule( required name, required type, thisModuleInvocationPath="", thisModulePath ){
+		var thisPath 			= arguments.thisModulePath 			?: variables[ arguments.type & "ModulesPath" ];
```

Comment:
@gpickin  I am coming back to this, but I fail to see why adding the invocation path and path are relevant? Can you explain?

Reply:
Because it needs to look deeper and deeper in the module tree.
Remember, this was to get these registered as ContentBox modules.

If we want only top level to register as ContentBox modules, we don't need the recursion.